File: Arrays/Python/Equal_Sum_and_Product.py
class Solution:
	def numOfsubarrays(self,arr, n):
    	# code here
    	summ = 0 
    	pr = 1
        k = n
        for i in range(n):
            summ = arr[i]
            pr = arr[i]
            for j in range(i+1, n):
                summ += arr[j]
                pr *= arr[j]
                if summ == pr:
                    k += 1
                    break
        return k

File: Solutions/Java/Equal_Sum_and_Product.java
class Solution {
    int numOfsubarrays(int[] arr, int n) {
        // code here
        int sum, pr;
        int k=n;
        for (int i=0; i<n; i++) {
            sum = arr[i];
            pr = arr[i];
            for (int j=i+1; j<n; j++) {
                sum += arr[j];
                pr *= arr[j];
                if (sum == pr) {
                    k++;
                    break;
                }
            }
        }
        return k;
    }
}
